opal/mca/common/ucx/common_ucx_wpool.c

```diff
@@ -308,6 +309,13 @@ opal_common_ucx_wpool_progress(opal_common_ucx_wpool_t *wpool)
             }
             opal_mutex_unlock(&winfo->mutex);
         }
+        if (active_workers == 0 && opal_list_get_size(&wpool->idle_workers)) {
+            // make sure to progress at least some
+            // no need to lock the winfo here, it is idle and we have the wpool mutex
+            _winfo_list_item_t *item = (_winfo_list_item_t*)opal_list_get_first(&wpool->idle_workers);
+            opal_common_ucx_winfo_t *winfo = item->ptr;
+            ucp_worker_progress(winfo->worker);
```

Comment:
Does it make sense to call opal progress instead?

Reply:
This is in the progress function that is called by `opal_progress`. It's important to progress UCX itself here.

Reply:
Can you elaborate more on why it is important to progress idle worker?

Reply:
Here is what I believe is going on: when I run with `-mca osc ucx` on a shared memory system, some operations require involving the target process, such as for AMOs used when locking a window. The target window is not aware of its window being locked, so all workers at the target are idle if the target does not have active operations going on itself. However, the AMO needs to be processed somehow (since it is not offloaded to any third-party entity such as the NIC) so occasionally the target process has to call into UCX to find the AMO request and process it. The only way to do so without unintended side effects was to progress an idle worker. There might be better ways to handle unexpected incoming RMA operations but I am not aware of them.

Reply:
Then you need to progress a worker that is being used as a target and not a random worker.
I think it just happens to be the first (and only) worker in the idle list.
But this may not be the case always.
See [[link](https://github.com/open-mpi/ompi/blob/master/ompi/mca/osc/ucx/osc_ucx_comm.c#L256)] for the worker I believe you really want to progress

Reply:
Got it, thanks! I changed the commit to progress the `dflt_worker` instead.